forte/data/data_store.py

```diff
@@ -937,7 +940,46 @@ def _parse_onto_file(self):
         """
```

Comment:
3. Does `files containing its parent entry` sound better? I gave an example after this sentence, maybe it will help understanding.
4. I am using `ontologyGenerator.parse_ontology_spec(
            self._onto_file_path, merged_scheme, merged_dict)` here, and this function will import the files along the ontology tree and store them in `merged_scheme`. When importing a class from file A, it will import every class in A.

Reply:
3. I actually (still) don't understand what you are trying to say here. What files? I didn't see any file operations in this function. Try to rethink/re-read what you write, and imagine if someone other than the developers can understand what is written.
4.
  - "import the files along the ontology", what do you mean by "import the files"? what file? Why are there files on the ontology tree?
  - No, this will only ensure that we import entry classes provided in the ontology definition, it does not have the ability to import everything under the same namespace.

Reply:
When the provided ontology spec file is `forte/ontology_specs/medical.json`, the output of `ontologyGenerator .parse_ontology_spec` would be every entry type in `ft.onto.base_ontology` plus type in `ftx.onto.clinical`, because "base_ontology.json" is in the imports path, which is because one of the type's parent_entry is in "base_ontology.json", as far as I understand. The function will recursively import every file in the `imports` path until it gets to the current file.

Reply:
Not sure if I understand now, but you can consider reusing the wording in the original docstring you used [here](https://github.com/asyml/forte/blob/master/forte/data/ontology/ontology_code_generator.py#L613).

Also, note that no matter what we do, we cannot guarantee that we import everything under a certain namespace, we can only import everything written in the specification file.

So now I see that you mean the "files" during the "import" section? It is actually not really easy to import them when running this function. The call of `ontologyGenerator.parse_ontology_spec` is safe. Since there are many caveats not considered. For example, where to import the other JSON files? Note that [this need to be set](https://github.com/asyml/forte/blob/master/forte/data/ontology/ontology_code_generator.py#L213) when using this code.

It is much harder to deal with all the imports at this point, so we should take one JSON schema file here only and use that. That's why I suggest using the [merged entry tree here](https://github.com/asyml/forte/blob/master/forte/data/ontology/ontology_code_generator.py#L685). Note that this is used in another place [here](https://github.com/asyml/forte/blob/642d57680c26fc1a36b31cb560e3f11785f9df11/forte/pipeline.py#L674). I am not sure if `merged entry tree` will give you all the classes (including the ones in top or core, but you should check that).

The process is simpler:
1. The user obtains a single ontology specification file that contains all the entry definitions he wanted to use, either manually or through the `-m` option described [here](https://asyml-forte.readthedocs.io/en/latest/toc/ontology_generation.html#ontology-generation-steps)
2. We take this one file and import all the types. In the docstring, write it very clear that we can only import the types in the file, remember, there is no way we can import everything under a certain namespace.


If you really want to use `ontologyGenerator.parse_ontology_spec`, then you need to either:
1. Make it very clear that we need to pass the import path (where the import ontology files will be placed). You can try to use a few default import paths (like the same directory), but throw exceptions if any one is not found.
2. Or make it very clear that we will not use any types not in the spec file provided (will not follow imports)